File: longest-common-subsequence/longest-common-subsequence.cpp
class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {
        int m=text1.size(),n=text2.size();
        int t[m+1][n+1];
        memset(t,0,sizeof(t));
        for(int i=1;i<m+1;i++)
        {
            for(int j=1;j<n+1;j++)
            {
                if(text1[i-1]==text2[j-1])
                    t[i][j]=t[i-1][j-1]+1;
                else 
                    t[i][j]=max(t[i-1][j],t[i][j-1]);
            }
        }
        return t[m][n];
        
    }
};

File: 1250-longest-common-subsequence/longest-common-subsequence.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        dp = [[-1]*len(text2) for _ in range(len(text1))]
        def solve(n,m):
            if n == len(text1) or m == len(text2):
                return 0
            if dp[n][m]!=-1:
                return dp[n][m]
            if text1[n] == text2[m]:
                dp[n][m] = solve(n+1,m+1)+1
            else:
                dp[n][m] = max(solve(n+1,m),solve(n,m+1))
            return dp[n][m]
        return solve(0,0)